scripts/deleteAndLoadSnapsets.js

```diff
@@ -0,0 +1,525 @@
+// A sample command that runs this script:
+// node deleteAndLoadSnapsets.js $COUCHDBURL $STATIC_DATA_DIR $BUILD_DATA_DIR
+//
+// There is also an optional final [--justDelete] argument for testing/debugging.
+// If present, the script exits with a zero exit status after deleting all the
+// snapset PrefsSafes and their GPII keys.  That is, the script does only the
+// first four steps listed above.
+
+"use strict";
+
+var http = require("http"),
+    url = require("url"),
+    fs = require("fs"),
+    fluid = require("infusion");
+
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.dataLoader");
+
+fluid.setLogging(fluid.logLevel.INFO);
+
+// Handle command line
+if (process.argv.length < 5) {
+    fluid.log("Usage: node deleteAndLoadSnapsets.js $COUCHDB_URL $STATIC_DATA_DIR $BUILD_DATA_DIR [--justDelete]");
+    process.exit(1);
+}
+
+/**
+ * Create a set of options for data loader and a function to retreive them.
+ * The options are based on the command line parameters and a set of database
+ * constants.
+ * @param {Array} processArgv - The command line arguments.
+ * @return {Object} - The options.
+ */
+gpii.dataLoader.initOptions = function (processArgv) {
+    var dbOptions = {};
+    dbOptions.couchDbUrl = processArgv[2];
+    dbOptions.staticDataDir = processArgv[3];
+    dbOptions.buildDataDir = processArgv[4];
+    if (processArgv.length > 5 && processArgv[5] === "--justDelete") { // for debugging.
+        dbOptions.justDelete = true;
+    } else {
+        dbOptions.justDelete = false;
+    }
+
+    // Set up database specific options
+    dbOptions.viewsUrl = dbOptions.couchDbUrl + "/_design/views";
+    dbOptions.prefsSafesViewUrl = dbOptions.couchDbUrl + "/_design/views/_view/findSnapsetPrefsSafes";
+    dbOptions.gpiiKeysViewUrl = dbOptions.couchDbUrl + "/_design/views/_view/findAllGpiiKeys";
+    dbOptions.parsedCouchDbUrl = url.parse(dbOptions.couchDbUrl);
+    dbOptions.staticData = [];
+    /* dbOptions.newViews; */
+    /* dbOptions.oldViews; */
+    dbOptions.snapsetPrefsSafes = [];
+    dbOptions.gpiiKeys = [];
+    dbOptions.postOptions = {
+        hostname: dbOptions.parsedCouchDbUrl.hostname,
+        port: dbOptions.parsedCouchDbUrl.port,
+        path: "/gpii/_bulk_docs",
+        auth: dbOptions.parsedCouchDbUrl.auth,
+        method: "POST",
+        headers: {
+            "Accept": "application/json",
+            "Content-Length": 0,          // IMPORTANT: FILL IN PER REQUEST
+            "Content-Type": "application/json"
+        }
+    };
+    fluid.log("COUCHDB_URL: '" +
+        dbOptions.parsedCouchDbUrl.protocol + "//" +
+        dbOptions.parsedCouchDbUrl.hostname + ":" +
+        dbOptions.parsedCouchDbUrl.port +
+        dbOptions.parsedCouchDbUrl.pathname + "'"
+    );
+    fluid.log("STATIC_DATA_DIR: '" + dbOptions.staticDataDir + "'");
+    fluid.log("BUILD_DATA_DIR: '" + dbOptions.buildDataDir + "'");
+
+    return dbOptions;
+};
+
+/*
```

Comment:
Turn this into a proper doc comment that gets linted. Add a special warning that the input argument will be modified.

Reply:
In terms of a warning, I've documented the properties added to the `options` input parameter in the function description.  It would be nice to be able to use some form of @return block tag here, but everything I tried failed to lint.